clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/HotRodStore.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.infinispan.subsystem.remote;
+
+import static org.jboss.as.clustering.infinispan.subsystem.remote.HotRodStoreConfiguration.REMOTE_CACHE_CONTAINER;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.commons.io.ByteBuffer;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.commons.persistence.Store;
+import org.infinispan.filter.KeyFilter;
+import org.infinispan.marshall.core.MarshalledEntry;
+import org.infinispan.persistence.TaskContextImpl;
+import org.infinispan.persistence.spi.AdvancedLoadWriteStore;
+import org.infinispan.persistence.spi.InitializationContext;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.util.KeyValuePair;
+import org.wildfly.clustering.infinispan.spi.RemoteCacheContainer;
+
+/**
+ * Simple implementation of Infinispan {@link AdvancedLoadWriteStore} configured with a started container-managed {@link RemoteCacheContainer}
+ * instance. Does not perform wrapping entries in Infinispan internal objects, this stores "raw" values.
+ *
+ * @author Radoslav Husar
+ */
+@Store(shared = true)
+public class HotRodStore<K, V> implements AdvancedLoadWriteStore<K, V> {
```

Comment:
Either we need to append the name of the embedded cache to the cache key - or we need to create a new remote cache using the name of the embedded cache based on the default cache config.

Reply:
Fixed with new Infinispan API org.infinispan.client.hotrod.RemoteCacheManagerAdmin#getOrCreateCache(java.lang.String, java.lang.String) defining a cache of the cache name using default configuration.